src/types.jl

```diff
@@ -23,31 +23,37 @@ end
 
 basic_free(b::Basic) = ccall((:basic_free_stack, :libsymengine), Void, (Ptr{Basic}, ), &b)
 
-function Basic(x::Clong)
+function Base.convert(::Type{Basic}, x::Clong)
     a = Basic()
     ccall((:integer_set_si, :libsymengine), Void, (Ptr{Basic}, Clong), &a, x)
     return a
 end
 
-function Basic(x::Culong)
+function Base.convert(::Type{Basic}, x::Culong)
     a = Basic()
     ccall((:integer_set_ui, :libsymengine), Void, (Ptr{Basic}, Culong), &a, x)
     return a
 end
 
-function Basic(x::BigInt)
+function Base.convert(::Type{Basic}, x::BigInt)
     a = Basic()
     ccall((:integer_set_mpz, :libsymengine), Void, (Ptr{Basic}, Ptr{BigInt}), &a, &x)
     return a
 end
 
-function Basic(s::String)
+function Base.convert(::Type{Basic}, s::String)
     a = Basic()
     ccall((:basic_parse, :libsymengine), Void, (Ptr{Basic}, Ptr{Int8}), &a, s)
     return a
 end
 
-Basic(ex::Expr) = Basic(string(ex))
+Base.convert(::Type{Basic}, ex::Union{Symbol,Expr}) = Basic(string(ex))
+function Base.convert(::Type{Basic}, x::Real)
```

Comment:
Can you replace `x::Real` with `x::Float64`? `Real` types with more precision should be converted into a `RealMPFR`.